vllm/engine/llm_engine.py

```diff
@@ -1784,8 +1783,8 @@ def _get_stats(self,
                                num_prompt_tokens_iter)
         # Spec decode, if enabled, emits specialized metrics from the worker in
         # sampler output.
-        if model_output and (model_output[0].spec_decode_worker_metrics
-                             is not None):
+        if model_output and isinstance(model_output[0], SamplerOutput) and (
+                model_output[0].spec_decode_worker_metrics is not None):
```

Comment:
Since spec decode isn't applicable to pooling models, I have removed `spec_decode_worker_metrics` from `PoolerOutput`. The type annotations that `model_output` is a list of `SamplerOutputs` is actually incorrect here (it can be a list of `PoolerOutput`) but I'm not bothered to fix it since we will probably rework this in V1 anyways.